core/src/main/java/org/kiwix/kiwixmobile/core/utils/ConnectivityReporter.kt

```diff
@@ -18,14 +18,22 @@
 
 package org.kiwix.kiwixmobile.core.utils
 
+import android.content.Context
+import android.content.Context.WIFI_SERVICE
 import android.net.wifi.WifiManager
 import java.lang.reflect.InvocationTargetException
 import java.lang.reflect.Method
 import javax.inject.Inject
 
-class ConnectivityReporter @Inject constructor(private val wifiManager: WifiManager) {
+class ConnectivityReporter @Inject constructor(private var wifiManager: WifiManager) {
 
-  fun checkWifi(): Boolean = wifiManager.isWifiEnabled && wifiManager.connectionInfo.networkId != -1
+  fun checkWifi(context: Context): Boolean {
+    wifiManager = context.applicationContext.getSystemService(WIFI_SERVICE) as WifiManager
```

Comment:
I don't understand it. why the two `wifiManager` are differents? We are creating the same instance type in the module as you did here.

```
 @Provides
  @KiwixScope
  fun provideWifiManager(context: Context): WifiManager =
    context.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
```

Reply:
I am not sure too, somehow if I initialize the variable there it works but it doesn't work with the old way, even though they are same maybe we are missing something

Reply:
update: if we are returning the statement then its working with old way

Reply:
@gouri-panda  
 seems like ``wifiManager.connectionInfo.networkId != -1`` is not giving true so the whole expression is getting false

Reply:
maybe rather than 
```
wifiManager.isWifiEnabled && wifiManager.connectionInfo.networkId != -1
```
we should go for either or?
```
wifiManager.isWifiEnabled || wifiManager.connectionInfo.networkId != -1
```

Reply:
Debug the two instances and check their properties such as `isWifiEnabled` and `connectionInfo.networkId`.

Reply:
NO. We should go with this
```
wifiManager.isWifiEnabled && wifiManager.connectionInfo.networkId != -1
```
Without one the point is meaningless

Reply:
okay I understood, I am not sure why but with active wifi connection my phone is giving ``wifiManager.connectionInfo.networkId = -1 `` 
but trying in emulator gives me ``wifiManager.connectionInfo.networkId = 0``
pretty weird

Reply:
okay I understood,
@gouri-panda  but I am not sure why but with active wifi connection my phone(android 11) and emulator with API 29 are giving ``wifiManager.connectionInfo.networkId = -1 `` 
but trying in emulator(API 26) gives me ``wifiManager.connectionInfo.networkId = 0``
pretty weird

Reply:
`getConnectionInfo()` was deprecated in Api level 31. WifiInfo available in `ConnectivityManager`. [Here more info](https://developer.android.com/reference/android/net/wifi/WifiManager#getConnectionInfo())

Reply:
oh, so now we maybe have the answer to why it was breaking!
what can be an ideal way to handle it now then?

Reply:
Documention shows how to do this. Try this by requesting NetworkRequest